src/main/java/model/Walker.java

```diff
@@ -54,19 +110,55 @@ public void move() {
             setX(getPrisonBubble().getX());
             setY(getPrisonBubble().getY());
         }
+
+        checkBounds();
+
         super.move();
     }
 
+    private void moveCollisionChecker() {
+        if (!levelController.causesCollision(getX(),
+                getX() + getWidth(),
+                getY() - calculateGravity(),
+                getY() + getHeight() - calculateGravity())) {
+            if (!jumping) {
+                setY(getY() - calculateGravity());
+            }
+            ableToJump = false;
+        } else {
+            if (!jumping) {
+                ableToJump = true;
+            }
+        }
+    }
+
+    /**
+     * This function returns the player if it is out of bounds.
+     */
+    private void checkBounds() {
+        if (getX() < walkerMinX) {
+            setX(walkerMinX);
+        } else if (getX() + getWidth() > walkerMaxX) {
+            setX(walkerMaxX - getWidth());
+        }
+
+        if (getY() < walkerMinY) {
+            setY(walkerMinY);
+        } else if (getY() + getHeight() > walkerMaxY) {
+            setY(walkerMaxY - getHeight());
+        }
+    }
+
```

Comment:
This looks like the code is inspired by the jumping of the player. This makes a lot of duplicate code, which should be put in GravityObject.